imap_processing/swapi/l1/swapi_l1.py

```diff
@@ -748,29 +748,47 @@ def swapi_l1(dependencies: ProcessingInputCollection) -> xr.Dataset:
             raise ValueError(
                 f"SWAPI SCI processing expected one L0 HK file. Found {len(hk_files)}."
             )
-        l1_hk_ds = load_cdf(hk_files[0])
+        l1b_hk_ds = load_cdf(hk_files[0])
         sci_dataset = process_swapi_science(
-            l0_unpacked_dict[SWAPIAPID.SWP_SCI], l1_hk_ds
+            l0_unpacked_dict[SWAPIAPID.SWP_SCI], l1b_hk_ds
         )
         return [sci_dataset]
 
     elif l0_unpacked_dict[SWAPIAPID.SWP_HK]:
         logger.info(f"Processing HK data for {l0_files[0]}.")
-        hk_ds = l0_unpacked_dict[SWAPIAPID.SWP_HK]
-        # Add HK datalevel attrs
+        # Get L1A and L1B HK data.
+        l1a_hk_data = l0_unpacked_dict[SWAPIAPID.SWP_HK]
+        l1b_hk_data = l0_unpacked_dict = packet_file_to_datasets(
+            l0_files[0], xtce_definition, use_derived_value=True
+        )[SWAPIAPID.SWP_HK]
```

Comment:
This seems odd to me to replace your l0_unpacked_dict after using it right before this. Do you need to update the dictionary?
```suggestion
        l1b_hk_data = packet_file_to_datasets(
            l0_files[0], xtce_definition, use_derived_value=True
        )[SWAPIAPID.SWP_HK]
```

Reply:
copy paste error :)